File: csharp/1108.defanging-an-ip-address.cs
/*
 * @lc app=leetcode id=1108 lang=csharp
 *
 * [1108] Defanging an IP Address
 */

using System.Collections.Generic;
using System.Linq;

// @lc code=start
public class Solution {

    private static IEnumerable<char> GetChars(string address) {
        foreach (char c in address) {
            if (c == '.') {
                yield return '[';
                yield return '.';
                yield return ']';
                continue;
            }
            yield return c;
        }
    } 

    public string DefangIPaddr(string address) {
        return new string(GetChars(address).ToArray());
    }
}
// @lc code=end

File: java/1108.defanging-an-ip-address.java
/*
 * @lc app=leetcode id=1108 lang=java
 *
 * [1108] Defanging an IP Address
 */

// @lc code=start
class Solution {
    public String defangIPaddr(String address) {
        int n = address.length();
        StringBuilder sb = new StringBuilder(n + 6);
        for (int i = 0; i < n; ++i) {
            char c = address.charAt(i);
            if (c == '.') {
                sb.append("[.]");
            } else {
                sb.append(c);
            }
        }
        return sb.toString();
    }
}
// @lc code=end

